### ISSUE TITLE
`default_branch` is reset to null when not set in .tf

### ISSUE THREAD

**Issue by abitrolly** [gitlabhq/terraform-provider-gitlab#299]
### Terraform Version
```
$ terraform -v
Terraform v0.12.24
+ provider.gitlab v2.5.0
```

### Affected Resource(s)
- gitlab_project

### Terraform Configuration Files
```hcl
provider "gitlab" {
}

resource "gitlab_project" "default" {
  name             = "dnf-go-gui"
  description      = "DNF What's New for upcoming package updates."
  visibility_level = "public"
}
```

### Expected Behavior
Expected [`default_branch` argument](https://www.terraform.io/docs/providers/gitlab/r/project.html#default_branch) to be optional. Running `terraform plan` should not reset `default_branch` for the project if it is not set in `.tf` file.

### Actual Behavior
```
...
        container_registry_enabled                       = true
      - default_branch                                   = "master" -> null
        description                                      = "DNF What's New for upcoming package updates."
...
```

### Steps to Reproduce
1. `terraform plan`

### References

https://github.com/terraform-providers/terraform-provider-gitlab/blob/a9f493f125f95ca4432c80e65f73faf36d326a83/gitlab/resource_gitlab_project.go#L38

**Comment by abitrolly** [comment]
The algorithm.
1. If default branch is not set in GitLab and not set in Terraform - do not store it in state
2. If default branch is not set in GitLab, but set in Terraform - apply Terraform version to state
3. If default branch is set in GitLab, but not set in Terraform - store it in state
4. If default branch is set in GitLab and set in Terraform - do usual stuff

**Comment by iyurev** [comment]
Good day! 
I have same problem with my configuration,  in my case:
1. I've created gitlab_project without definition default_branch
2. After master branch became in the project , terraform apply is trying  change it to -> null and 
3. Failed with error:
```400 {message: {base: [Could not change HEAD: branch '' does not exist]}}```

**Comment by abitrolly** [comment]
I never tried to apply it. I guess the issue is more serious then. I got development environment ready for the fix. What I do not know, is how to use Terraform API to check that `default_branch` attribute for `gitlab_project` is defined in .tf file?

**Comment by iyurev** [comment]
I didn’t read provider development guide.(
But seems we can catch it here: https://github.com/terraform-providers/terraform-provider-gitlab/blob/master/gitlab/resource_gitlab_project.go

**Comment by abitrolly** [comment]
The default value in GitLab API is `master`. https://docs.gitlab.com/ee/api/projects.html#edit-project

Need to check what API returns exactly. Is it an empty string, missing value or some kind of `null` value.

**Comment by abitrolly** [comment]
For projects without repo API https://gitlab.com/api/v4/projects returns `"default_branch": null`. Not empty string. Need to check what the type Go converts to.

For existing repos it is impossible to "unselect" Default branch from web interface.

![image](https://user-images.githubusercontent.com/8781107/85100957-07e3ad00-b20a-11ea-8c2e-7a7b1062ccd2.png)

Therefore the operation to set `default_branch` to `null` doesn't make sense and should be ignored. The 400 error is shown anyway https://github.com/terraform-providers/terraform-provider-gitlab/issues/299#issuecomment-627270803

**Comment by abitrolly** [comment]
When creating new project, I see this in debug log.
```
2020/06/19 20:55:55 [WARN] Provider "registry.terraform.io/-/gitlab" produced an unexpected new value for gitlab_project.default, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .remove_source_branch_after_merge: was null, but now cty.False
      - .description: was null, but now cty.StringVal("")
      - .path: was null, but now cty.StringVal("terraform-empty-project")
      - .default_branch: was null, but now cty.StringVal("")
gitlab_project.default: Creation complete after 1s [id=19484561]

```

**Comment by abitrolly** [comment]
Made a fix in #343. You can checkout branch, `go build` it and copy the binary into your project directory. After `terraform init` it should work ok.

**Comment by FernandoMiguel** [comment]
not sure if related https://github.com/terraform-providers/terraform-provider-gitlab/issues/349
